coder-sdk/client.go

```diff
@@ -14,15 +18,38 @@ const Me = "me"
 
 // ClientOptions contains options for the Coder SDK Client.
 type ClientOptions struct {
-	// BaseURL is the root URL of the Coder installation.
+	// BaseURL is the root URL of the Coder installation (required).
 	BaseURL *url.URL
 
 	// Client is the http.Client to use for requests (optional).
+	//
 	// If omitted, the http.DefaultClient will be used.
 	HTTPClient *http.Client
 
-	// Token is the API Token used to authenticate
+	// Token is the API Token used to authenticate (optional).
+	//
+	// If Token is provided, the DefaultClient will use it to
+	// authenticate. If it is not provided, it will require another
+	// type of credential, such as a Email/Password pair.
 	Token string
+
+	// Email will be used to authenticate.
+	//
+	// DefaultClient requires exactly one type of credential
+	// (Token, or Email/Password pair).
+	//
+	// NewClient will exchange the Email and Password for a Token
+	// and will not retain them in-memory after NewClient returns.
+	Email string
+
+	// Password will be used to authenticate.
+	//
+	// DefaultClient requires exactly one type of credential
+	// (Token, or Email/Password pair).
+	//
+	// NewClient will exchange the Email and Password for a Token
+	// and will not retain them in-memory after NewClient returns.
+	Password string
```

Comment:
Should document here that email and password is only supported for built in auth, not oidc.